credential-provider/src/controllers/core.controller.ts

```diff
@@ -32,8 +37,20 @@ export class CoreController {
   }
 
   @Get('2fa')
-  @Render('2fa')
-  public async render2faPage() {
-    return;
+  public async render2faPage(@Req() req: Request, @Res() res: Response) {
+    /**
+     * Renderer for interaction-hook page
+     *
+     * 1. Extracts session_token from query param
+     * 2. Calls CoreService.createResponseToken({ session_token }) -> callbackUrl
+     * 3. Embed callbackUrl into object for template to map
+     * 4. Render template when callbackUrl is created
+     */
+    const session_token = req.query.session_token as string;
+    this.logger.warn(`Extracted session_token --> ${session_token}`);
+    const callbackUrl = await this.coreService.createResponseToken({
```

Comment:
createResponseToken returns a callbackUrl?

Reply:
Changed the function name to `createCallbackUrl`